### PR TITLE
DellEMC: Z9332f fix platform issue

### PR DESCRIPTION
**Pull request by aravindmani-1**
#### Why I did it

- Following I2C errors were seen in Z9332f platform during PMON docker startup. 

[    4.211968] i2c i2c-22: new_device: Instantiated device optoe3 at 0x50
[   63.803230] i2c i2c-22: delete_device: Deleting device optoe3 at 0x50
[   63.804691] sysfs: cannot create duplicate filename '/devices/pci0000:00/0000:00:1c.0/0000:09:00.0/goodcores-i2c.3/i2c-3/i2c-22/22-0050'
[   63.818497] i2c i2c-22: Failed to register i2c client optoe1 at 0x50 (-17) 

#### How I did it

- This occurs during OPTOE driver re-initialization where drivers will be changed based on the transceiver type.
- Add a small amount of delay after EEPROM driver deletion.

#### How to verify it

- Do cold boot/fast boot/warmboot and check kernel logs.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

### CODE REVIEW

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/sfp.py [hunk 1]
```diff
@@ -1022,16 +1022,19 @@ def reinit_sfp_driver(self):
             #Avoid re-initialization of the QSFP/SFP optic on QSFP/SFP port.
             if (self.media_type == 'SFP' and (driver_name == 'optoe1' or driver_name == 'optoe3')):
                 subprocess.Popen(delete_device, shell=True, stdout=subprocess.PIPE)
+                time.sleep(0.2)
```
Comment (lguohan) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/sfp.py:1025]: what is the use case here? why is that pmon restart will reinit sfp driver?

Comment (aravindmani-1) [reply platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/sfp.py:1025]: Initially, optoe3 driver will be loaded for all the ports in z9332f platform init script.
If 100G is inserted, EEPROM driver has to be changed to optoe1.
optoe1 ===> QSFP+,QSFP28
optoe2 ===> SFP+,SFP28
optoe3 ===> CMIS devices.

As optoe3 is attached for all the transceivers, pmon will reinitialize the EEPROM driver when there is a mismatch.

**Comment by aravindmani-1** [comment]
@lguohan , @jleveque can you please review this PR?.

**Comment by aravindmani-1** [comment]
Addressed this PR changes as part of https://github.com/Azure/sonic-buildimage/pull/7457

**Pull request closed without merging**